module/zfs/vdev_draid.c

```diff
@@ -1150,6 +1150,16 @@ vdev_draid_min_alloc(vdev_t *vd)
 	return (vdc->vdc_ndata << vd->vdev_ashift);
 }
 
+static uint64_t
+vdev_draid_worst_alloc(vdev_t *vd)
+{
+	vdev_draid_config_t *vdc = vd->vdev_tsd;
+
+	ASSERT3P(vd->vdev_ops, ==, &vdev_draid_ops);
+
+	return (vdc->vdc_groupwidth);
```

Comment:
There is no reason to combine the worst ashift with the "worst" geometry when they can be in different tvds. Instead here just have

> `return (vdc->vdc_groupwidth << vd->vdev_ashift);`

Similarly for `vdev_raidz_worst_alloc`. Then in `spa_get_worst_case_asize` just do

> `inflation = roundup(lsize, spa->spa_worst_alloc);`

Reply:
I apologize for being dense. I understand what you are getting at now, where the raidz has the "bad" geometry but a normal ashift, and the mirror has the high ashift but no geometry inflation.

I'll rework it to handle this case as you suggested.

Reply:
Hold your horses. Both of our logic are wrong. Writing 1MB to a 4 wide raidz3 will inflate it to 4MB. None of our roundups are going cover that case. This needs some more involved math.

Reply:
Looking more closely into this, it is doing the comparisons against the quota and dataset layer free space, which has already been "deflated" to account for parity etc. So I think we don't need to account for 4x for 1MB of logical data written to a RAID-Z2.

Reply:
Are you planning to make this change, of having each vdev calculate its minimum allocation size in bytes, taking into account ashift?